src/Dto/Segments/CreateSegment.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Dto\Segments;
+
+use OneSignal\Dto\AbstractDto;
+
+class CreateSegment implements AbstractDto
+{
+    /**
+     * @var non-empty-string
+     */
+    protected string $name;
+
+    /**
+     * @var array<int, array>
+     */
+    protected array $filters = [];
+
+    /**
+     * @param non-empty-string  $name
+     * @param array<int, array> $filters
+     */
+    public function __construct(string $name, array $filters = [])
+    {
+        $this->name = $name;
+        $this->filters = $filters;
+    }
+
+    /**
+     * @param non-empty-string $name
+     */
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * @param array<int, array> $filters
+     */
+    public function setFilters(array $filters = []): self
```

Comment:
I am thinking for filters disvoery/autocompletion:
 - Introduce FilterDTO: { $field; $relation (LT/GT/EQUAL/NOT_EQUAL); $value; } and FilterConditionDTO: { $operator (AND/OR); }
 - For `FilterDTO $field` we could typehint all available values from `https://documentation.onesignal.com/reference/create-notification#send-to-users-based-on-filters`
 - typehint as `@param list<FilterDTO|FilterConditionDTO>`
 - add `addFilter(FilterDTO|FilterConditionDTO)` method